sidecar/supervisor/appsupervisor.go

```diff
@@ -204,6 +198,8 @@ func terminateSubprocesses(procs []*process, sig os.Signal) {
 		time.Sleep(100 * time.Millisecond)
 	}
 
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Just giving any processes who were forcibly killed the chance to exit - modified the placement of this sleep to be in the same block as the timeout logic and added some comments